lib/phoenix_live_view/channel.ex

```diff
@@ -120,17 +139,54 @@ defmodule Phoenix.LiveView.Channel do
     end
   end
 
-  defp handle_result({:noreply, %Socket{} = new_socket}, _fa, state) do
-    {:noreply, push_render(state, new_socket)}
+  defp mount_handle_params(uri, channel_params, state) do
+    if function_exported?(state.socket.view, :handle_params, 2) do
+      case View.live_link_info!(state.socket, uri) do
+        {:internal, params} ->
+          channel_params
+          |> Map.merge(params)
+          |> view_module(state).handle_params(state.socket)
```

Comment:
We need to check the `:external` branch here.

Reply:
I believe in this case, it will always be internal because the client should only join with the uri that belongs to that LV. I will change it to a match instead of case